### PR TITLE
Feature/add more tensor from string tests

### PR DESCRIPTION
**Pull request by markos**
Make Tensor::FromString more robust, add more tests

### CODE REVIEW

File: libs/math/include/math/tensor.hpp [hunk 1]
```diff
@@ -707,8 +707,10 @@ Tensor<T, C> Tensor<T, C>::FromString(byte_array::ConstByteArray const &c)
     UNSET,
     COLON,
     NEWLINE
-  } new_row_marker         = UNSET;
-  bool reached_actual_data = false;
+  } new_row_marker           = UNSET;
+  bool   reached_actual_data = false;
+  size_t first_row_size      = 0;
```
Comment (vhavrylov-ncube) [inline libs/math/include/math/tensor.hpp]: std::size_t

File: libs/math/tests/unit/tensor/constructor.cpp [hunk 2]
```diff
@@ -37,6 +37,144 @@ TYPED_TEST(TensorConstructorTest, string_construction)
...
   ASSERT_EQ(tensor.At(0, 2), TypeParam(4));
+
+  auto tensor1 = fetch::math::Tensor<TypeParam>::FromString(R"(
+  	1 3 4 4.1
+  	5 6 7 7.999
+  	8 9 10 11.11111
+  )");
+  ASSERT_EQ(tensor1.shape().at(0), 3);
+  ASSERT_EQ(tensor1.shape().at(1), 4);
+  ASSERT_EQ(tensor1.At(0, 0), math::Type<TypeParam>("1"));
```
Comment (vhavrylov-ncube) [inline libs/math/tests/unit/tensor/constructor.cpp]: (optional) ASSERT_EQ will immediately kill the test if failed, but each comparison here and in next line is actually not so critical - if it fails, we still could check the rest of elements; so I would suggest EXPECT_EQ for element comparisons.

File: libs/math/include/math/tensor.hpp (hunk 1 above)
Comment (markos) [reply libs/math/include/math/tensor.hpp]: fixed

File: libs/math/tests/unit/tensor/constructor.cpp (hunk 2 above)
Comment (markos) [reply libs/math/tests/unit/tensor/constructor.cpp]: replaced with EXPECT

**Comment by markos** [comment]
Closing because it's dealt in #2162

**Pull request closed without merging**